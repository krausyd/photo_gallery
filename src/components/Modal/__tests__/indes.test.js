import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Modal from '..';

const mockCloseModal = jest.fn();
const mockCurrentPhoto = {
    name: 'Pancakes',
    category: 'food',
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie',
    index: 0,
};

afterEach(cleanup);

describe('Modal tests', () => {
    it('renders', () => {
        render(<Modal onClose={mockCloseModal} currentPhoto={mockCurrentPhoto} />);
    });

    it('looks like snapshot', () => {
        const { asFragment } = render(<Modal onClose={mockCloseModal} currentPhoto={mockCurrentPhoto} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('closes the modal', () => {
        const { getByText } = render(<Modal onClose={mockCloseModal} currentPhoto={mockCurrentPhoto} />);
        fireEvent.click(getByText('Close this modal'));
        expect(mockCloseModal).toHaveBeenCalledTimes(1);
    });
});