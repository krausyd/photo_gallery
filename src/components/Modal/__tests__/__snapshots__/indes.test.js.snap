// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal tests looks like snapshot 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
        Pancakes
      </h3>
      <img
        alt="food"
        src="0.jpg"
      />
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie
      </p>
      <button
        type="button"
      >
        Close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
